src/main/java/com/mraof/minestuck/blockentity/machine/PunchDesignixBlockEntity.java

```diff
@@ -152,16 +155,16 @@ private void handleKeyboardClick(Player player)
 			}
 		} else
 		{
-			if(AlchemyHelper.isReadableCard(heldStack))
+			if(heldStack.is(MSItems.CAPTCHA_CARD) && !heldStack.has(MSItemComponents.ENCODED_ITEM))
```

Comment:
The statement of "itemstack is a captcha card && does not have encoded item compnent" appears several times in code, perhaps it would make for a good static method in EncodedItemComponent!

Reply:
Added it as CaptchaCardItem.isUnpunchedCard()